loader/loader.go

```diff
@@ -460,7 +460,20 @@ func (l *Loader) Init(ctx context.Context) (err error) {
 	dbCfg.RawDBCfg = config.DefaultRawDBConfig().
 		SetMaxIdleConns(l.cfg.PoolSize)
 
-	l.toDB, l.toDBConns, err = createConns(tctx, l.cfg, l.cfg.PoolSize)
+	// used to change loader's specified DB settings, currently SQL Mode
+	lcfg, err := l.cfg.Clone()
+	if err != nil {
+		return err
+	}
+	lowerM := map[string]struct{}{}
+	for k := range l.cfg.To.Session {
+		lowerM[strings.ToLower(k)] = struct{}{}
+	}
+	if _, ok := lowerM["sql_mode"]; !ok {
```

Comment:
Is a `bool` value like `hasSQLMode` enough rather than a `map`? (check `strings.ToLower(k) == sql_mode` in the `for` loop)

Reply:
https://github.com/pingcap/dm/pull/1134/commits/7fb7fdc2aca8695f0878d1ff98d274f1046fb693